### PR TITLE
Upgrade to .NET Core 2.1

### PR DESCRIPTION
**Pull request by shhsu**

### CODE REVIEW

File: build/version.props [hunk 1]
```diff
@@ -2,5 +2,26 @@
 
   <PropertyGroup>
     <IISAdministrationVersion>2.2.1</IISAdministrationVersion>
+    <MicroBuildCoreVersion>0.3.0</MicroBuildCoreVersion>
+    <SystemSecurityCryptographyX509CertificatesVersion>4.3.2</SystemSecurityCryptographyX509CertificatesVersion>
+    <SerilogExtensionsLoggingVersion>2.0.2</SerilogExtensionsLoggingVersion>
+    <SerilogVersion>2.7.1</SerilogVersion>
```
Comment (shirhatti) [inline build/version.props]: Did you do an OSS registration for all the updated third-party dependencies?

Comment (shhsu) [reply build/version.props]: I did not do it myself. I am going to double check these OSS dependencies

File: test/Microsoft.IIS.Administration.Tests/DefaultDocument.cs [hunk 2]
```diff
@@ -141,7 +141,7 @@ public static JArray GetFiles(HttpClient client, JObject docFeature)
             return docFeature.Value<JArray>("files");
         }
 
-        public static JObject CreateFile(HttpClient client, JObject docFeature, string fileName)
+        public static JObject CreateFileInternal(HttpClient client, JObject docFeature, string fileName)
```
Comment (jimmyca15) [inline test/Microsoft.IIS.Administration.Tests/DefaultDocument.cs]: Not really internal if its public.

File: build/version.props [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
   <PropertyGroup>
-    <IISAdministrationVersion>2.2.1</IISAdministrationVersion>
+    <IISAdministrationVersion>3.0.0</IISAdministrationVersion>
```
Comment (jimmyca15) [inline build/version.props]: @shirhatti the thinking here was that upgrading to a .NET Core 2.1 dependency was a big enough change to warrant the major version update. Thoughts ?

Comment (shhsu) [reply build/version.props]: talked to @bariscaglar and @shirhatti , decided on 2.3.0

**Pull request merged**